src/test/java/com/google/sps/email/EmailTesting.java

```diff
@@ -0,0 +1,173 @@
+
+    private static final String GOOD_DEED = "GoodDeed";
+    private static final String NAME = "Name";
+    private static final String DESCRIPTION = "Description";
+    private static final String POSTED_YET = "Posted Yet";
+    private static final String DAILY_DEED = "Daily Deed";
+    private static final String TIMESTAMP = "Timestamp";
+    private static final String LINK = "Link";
+
+    private static final String TITLE = "Deed Name";
+    private static final String DESCRIPTION_INPUT = "Deed Description";
+    private static final String TRUE_STRING = "true";
+    private static final String FALSE_STRING = "false";
+    private static final long TIMESTAMP_INPUT = 67890;
+    private static final boolean TRUE = true;
+    private static final boolean FALSE = false;
+
+    private final LocalServiceTestHelper helper = 
+        new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+    @Before
+    public void setUp() {
+        emailServlet = new EmailServlet();
+        helper.setUp();
+        MockitoAnnotations.initMocks(this);
+        Mailbox.clearAll();
+    }
+
+    @Test
+    public void testFetchEmails() {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+
+        Entity testEntity = new Entity(USER);
+        testEntity.setProperty(EMAIL, EMAIL_INPUT);
+        ds.put(testEntity);
+
+        Query query = new Query(USER);
+
+        Assert.assertEquals(1, ds.prepare(query).countEntities(withLimit(10)));
+
+        List<String> emails = emailServlet.fetchEmails();
+        String email = emails.get(0);
+
+        Assert.assertTrue(email.equals(EMAIL_INPUT));
+    }
+
+    @Test
+    public void testsendEmail() throws MessagingException, IOException, MessagingException {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+        
+        Entity testEntity = new Entity(GOOD_DEED);
+        testEntity.setProperty(NAME, TITLE);
+        testEntity.setProperty(DESCRIPTION, DESCRIPTION_INPUT);
+        testEntity.setProperty(POSTED_YET, FALSE_STRING);
+        testEntity.setProperty(DAILY_DEED, FALSE_STRING);
+        testEntity.setProperty(TIMESTAMP, TIMESTAMP_INPUT);
+        testEntity.setProperty(LINK, LINK);
+        ds.put(testEntity);
+
+        GoodDeed deed = 
+            new GoodDeed(testEntity.getKey(), testEntity.getKey().getId(), TITLE, DESCRIPTION_INPUT, FALSE, TIMESTAMP_INPUT, LINK);
+        
+        String content = deed.getTitle() + ":\n" + deed.getDescription();
+        
+        emailServlet.sendEmail(deed, EMAIL_INPUT);
+
+        List<Message> inbox = Mailbox.get(EMAIL_INPUT);
+  
+        Assert.assertTrue(inbox.size() == 1);  
+        Assert.assertEquals(EMAIL_SUBJECT, inbox.get(0).getSubject());
+        Assert.assertEquals(content, inbox.get(0).getContent());
+    }
+
+    @Mock
+    HttpServletRequest request;
+    @Mock
+    HttpServletResponse response;
```

Comment:
these can be moved to within the method `testDoGet()` as we are using them elsewhere. You can just simply do `emailServlet.doGet(mock(), mock());`. 

Additionally you can remove the `MockitoAnnotations.initMocks(this);` line with this change

Reply:
@abhijitnukalapati I'm not sure if I understand what you mean...